libraries/AP_HAL_ChibiOS/hwdef/scripts/chibios_hwdef.py

```diff
@@ -3135,6 +3140,11 @@ def add_iomcu_firmware_defaults(f):
 #define AP_INERTIALSENSOR_ENABLED 0
 #endif
 
+// no RC_Channels library:
+#ifndef AP_RC_CHANNEL_ENABLED
```

Comment:
I just don't get this name - feels like we could come up with something more understandable

Reply:
The comment just above it doesn't clarify?

The comment should read `RC_Channel`  not `RC_Channels`

Reply:
I am clearly asking this wrong :-P What is this PR doing? Like what is it removing? It's not removing RC channels - how can you have an RC protocol without RC channels? I can see it might be removing the RC_Channels library - but what does that actually mean? I suspect it means you are removing RC channel *function* support.

Reply:
Ah, sorry.  The context has all been elsewhere!

One of our partners asked about receiving RC via a DroneCAN AP_Periph.  So decoding these protocols on the device, sending packed 11-bit values (or whatever...) over to the main autopilot in new CAN packets.

Much like the IOMCU's decoding, we don't want RC_Channel involved - and you can actually see that the changes in `AP_RCProtocol.cpp` show that.

So no, we are actually removing the RC_Channel *library* support :-)  AP_Periph just compiles more libraries, thus the changes in some other libraries.

This PR is based off the - what? - 5 other PRs I've done around RC recently, and compiles while instantiating AP_RCProtocol in AP_Periph: https://github.com/peterbarker/ardupilot/tree/pr/ap-periph-rcprotocol